opm/autodiff/BlackoilWellModel.hpp

```diff
@@ -126,44 +115,59 @@ namespace Opm {
 
             // using the solution x to recover the solution xw for wells and applying
             // xw to update Well State
-            void recoverWellSolutionAndUpdateWellState(const BVector& x, WellState& well_state) const;
+            void recoverWellSolutionAndUpdateWellState(const BVector& x);
 
-            int numWells() const;
+            // Check if well equations is converged.
+            bool getWellConvergence(const std::vector<Scalar>& B_avg) const;
 
-            /// return true if wells are available in the reservoir
-            bool wellsActive() const;
+            // return all the wells.
+            const WellCollection& wellCollection() const;
+            WellCollection& wellCollection();
```

Comment:
This non-const method should not need to be public.

Reply:
It seems that the call to 
wellModel().wellCollection().groupControlActive()
and
wellModel().wellCollection().groupTargetConverged(wellModel().wellState().wellRates()
in #250 in BlackoilModelEbos whant's non const wellCollections.

Reply:
That is very strange, as both those methods of WellCollection are const! There must be something else preventing this. Perhaps there is no const overload of wellModel() in the BlackoilModelEbos class?